amqp-connector/src/main/java/de/uni_stuttgart/tik/viplab/websocket_api/amqp/AMQPConnector.java

```diff
@@ -38,4 +46,19 @@
 		});
 	}
 
+	@Incoming("results")
+	public CompletionStage<Void> processResults(Message<String> message) {
+		try {
+			ComputationResult result = jsonb.fromJson(message.getPayload(), ComputationResult.class);
+			ComputationResultMessage resultMessage = new ComputationResultMessage(result);
+			notificationService.notify("computation:" + result.computation, session -> {
+				session.send(resultMessage);
+			});
+			return message.ack();
+		} catch (Exception e) {
+			e.printStackTrace();
+			return message.ack();
```

Comment:
Yes, but we need to improve the handling of broken messages. I don't see any point on leaving broken messages on the broker.I would propose creating an task that we need to inform the client that a message for him could not be processed.

Reply:
Depending on the cause of the error they should be handled differently. If it's a error by the websocket client then it should be reported to them, these errors maybe can already be detected by the websocket api. For the other errors we should use an invalid message queue with monitoring. So an admin can look at the invalid message and then decide what to do, e.g. debug the cause and fix the code.

So the catch all case should be sending the message to a invalid message queue, log an error and ack the message.

Reply:
For proper handling, I would suggest the following. 
- Message sent by the client that are broken could be directly reported back to client. This case I think is easy, as it would be a direct response to a message from the client
- In the case we have here, we are dealing with response of a computation. I would propose that the websocket-api in this case at least tries if e.g. the message is JSON at all, and if it is tries to extract the computation-id to which it belongs and then reports an error back. If it either not JSON or does not contain a computation-id, there is nothing we can do to inform the client waiting for a response that something is wrong.

I agree with logging the error and acknowledging the message. But I would prefer not to send the data back to the broker. I would prefer a solution where the websocket-api itself stores the message on a disk and then utilize the metrics-api to track the number of messages that are recorded there. This could then easily be monitored. Since this is mostly a feature used during the development of a backend, I would even suggest adding a flag to only add a log entry and then discard the message.